servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ServerDuplexHandler.java

```diff
@@ -134,17 +135,17 @@ private NettyH2HeadersToHttpHeaders h2HeadersToH1HeadersServer(Http2Headers h2He
         h2Headers.remove(Http2Headers.PseudoHeaderName.SCHEME.value());
         h2HeadersSanitizeForH1(h2Headers);
         if (httpMethod != null) {
-            final Long contentLength = h2Headers.getLong(CONTENT_LENGTH);
+            final boolean containsContentLength = h2Headers.contains(CONTENT_LENGTH);
             if (clientMaySendPayloadBodyFor(httpMethod)) {
-                if (contentLength == null) {
-                    if (fullRequest) {
+                if (!containsContentLength) {
+                    if (fullRequest && shouldAddZeroContentLength(httpMethod)) {
                         h2Headers.set(CONTENT_LENGTH, ZERO);
                     } else {
                         h2Headers.add(TRANSFER_ENCODING, CHUNKED);
                     }
                 }
-            } else if (contentLength >= 0L) {
-                throw new IllegalArgumentException("content-length (" + contentLength +
+            } else if (containsContentLength) {
+                throw new IllegalArgumentException("content-length (" + h2Headers.getLong(CONTENT_LENGTH) +
```

Comment:
just do `h2Headers.get(CONTENT_LENGTH)` bcz we are using it as a string here anyways.